stages/org.osbuild.sysconfig

```diff
@@ -100,17 +194,65 @@ def configure_network(tree, network_options):
     return 0
 
 
+def configure_network_scripts(tree, network_scripts_options):
+    if not network_scripts_options:
+        return 0
+
+    # if the network-scripts dir does not yet exist create it
+    os.makedirs(f"{tree}/etc/sysconfig/network-scripts", exist_ok=True)
+
+    network_scripts_ifcfg_options = network_scripts_options.get("ifcfg", {})
+
+    if configure_network_scripts_ifcfg(tree, network_scripts_ifcfg_options):
+        return 1
+
+    return 0
+
+
+def configure_network_scripts_ifcfg(tree, network_scripts_ifcfg_options):
+    for ifname, ifconfig in network_scripts_ifcfg_options.items():
+        lines = []
+        for option, value in ifconfig.items():
+            if option == "device":
+                lines.append(f'DEVICE="{value}"\n')
+            elif option == "bootproto":
+                lines.append(f'BOOTPROTO="{value}"\n')
+            elif option == "onboot":
+                lines.append(f'ONBOOT="{bool_to_string(value)}"\n')
+            elif option == "type":
+                lines.append(f'TYPE="{value}"\n')
+            elif option == "userctl":
+                lines.append(f'USERCTL="{bool_to_string(value)}"\n')
+            elif option == "peerdns":
+                lines.append(f'PEERDNS="{bool_to_string(value)}"\n')
+            elif option == "ipv6init":
+                lines.append(f'IPV6INIT="{bool_to_string(value)}"\n')
+            else:
+                # schema does not currently allow any additional properties but it may at some point
+                print(f"Error: unknown property {option} specified for sysconfig network-scripts/ifcfg config.")
+                return 1
```

Comment:
Just throw a `ValueError` exception, saves you the check where you call the function as well, and in the future we might catch it and properly report it to the user instead of just writing it to the logs. (Should have done that in the other functions as well, heh)

Reply:
Would you mind if I change all `configure_*` functions to raise the exception and remove all checks of the return value as part of this change?

Reply:
Would love it!

Reply:
OK, done :sunglasses: